packages/camera/camera_android_camerax/CONTRIBUTING.md

```diff
@@ -0,0 +1,46 @@
+# Contributing to camera\_android\_camerax
+
+## Plugin structure
+
+The `camera_platform_interface` implementation is located at
+`lib/src/android_camera_camerax.dart`, and it is implemented using Dart classes that
+are wrapped versions of native Android Java classes.
+
+In `lib/src/`, you will find all of the Dart-wrapped native classes that the plugin
+currently uses in its implementation. Each of these classes uses an `InstanceManager`
+(implementation in `instance_manager.dart`) to manage objects that are created by
+the plugin implementation that map to objects of the same type created on the native
+side. This plugin uses [`pigeon`][1] to generate the communication layer between Flutter
+and native Android code, so each of these Dart-wrapped classes also have Host API and
+Flutter API implementations, as needed. The communication interface is defined in
+the `pigeons/camerax_library.dart` file. After editing the communication interface,
+regenerate the communication layer by running
+`dart run pigeon --input pigeons/camerax_library.dart` from the plugin root.
+
+On the native side in `android/src/main/java/io/flutter/plugins/camerax/`, you'll
+find the Host API and Flutter API implementations of the same classes wrapped with
+Dart in `lib/src/`. These implementations call directly to the classes that they 
+are wrapping in the CameraX library or other Android libraries. The objects created
+in the native code map to objects created on the Dart side, and thus, are also
+managed by an `InstanceManager` (implementation in `InstanceManager.java`).
+
+If you need to access any Android classes to contribute to this plugin, you should
+search in `lib/src/` for any Dart-wrapped classes you may need. If any classes that
+you need are not wrapped or you need to access any methods not wrapped in a class,
+you must take the additional steps to wrap them to maintain the structure of this plugin.
+
+For more information on the approach of wrapping native libraries for plugins, please
```

Comment:
Having read though this I think a bunch of the information only makes sense after understanding the architecture. A plain english version should be at the top along with a link to the design document.

Reply:
Oh and @bparrishMines to help people ask questions and spread this pattern or even just write about it you should come up with a unique name for the pattern.